integration/test_collection.py

```diff
@@ -879,3 +880,44 @@ def test_empty_search_returns_everything(client: weaviate.Client):
     assert objects[0].metadata.creation_time_unix is not None
 
     client.collection.delete("TestReturnEverything")
+
+
+def test_insert_date_property(client: weaviate.Client):
+    collection = client.collection.create(
+        CollectionConfig(
+            name="TestInsertDateProperty",
+            vectorizer=Vectorizer.NONE,
+            properties=[Property(name="date", data_type=DataType.DATE)],
+        )
+    )
+
+    now = datetime.now(timezone.utc)
+    uuid = collection.data.insert(data={"date": now})
+
+    obj = collection.data.get_by_id(uuid)
+    assert obj.data["date"] == now.isoformat(sep="T", timespec="milliseconds").replace(
```

Comment:
should we also automatically parse this as a datetime?

Reply:
Not for the non-ORM API without generics but definitely for generics and the ORM